machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,747 @@
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// MachineConfigRoleLabelKey is metadata key in the MachineConfig. Specifies the node role that config should be applied to.
+// For example: `master` or `worker`
+const MachineConfigRoleLabelKey = "machineconfiguration.openshift.io/role"
+
+// KubeletConfigRoleLabelPrefix is the label that must be present in the KubeletConfig CR
+const KubeletConfigRoleLabelPrefix = "pools.operator.machineconfiguration.openshift.io/"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControllerConfig describes configuration for MachineConfigController.
+// This is currently only used to drive the MachineConfig objects generated by the TemplateController.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
```

Comment:
worth putting a "can change at any time for any reason" api into v1alpha1, instead of putting it in `machineconfiguration/v1/`?

Reply:
It probably would if we were just now introducing ControllerConfig. We talked about this in control plane lifecycle arch on 9/19/2023, and my use of level 4 was probably erroneous.  We're just going to mark it level 1 and keep it in v1 since it's already out there.